x-pack/plugins/stack_connectors/server/connector_types/webhook/index.ts

```diff
@@ -54,6 +55,25 @@ const configSchemaProps = {
   }),
   headers: nullableType(HeadersSchema),
   hasAuth: schema.boolean({ defaultValue: true }),
```

Comment:
`hasAuth` seems to be a property common to several connector types, attempting to remove it broke a lot of tests at a deeper level.

It would also be a major breaking change for existing connectors, I'm not quite sure how to handle that.

Reply:
It looks like `hasAuth` used by email and cases, but should just be a config, so independent of any other connectors.

But I'm not suggesting we remove it - though that IS a possibility - change 2-state `hasAuth` to 3-state `authType`.  That would require a migration, so ... not a happy path for me :-)

My suggestion is to remove the NONE choice of `authType`.  Since that's already covered by `hasAuth`.  So you could only set `authType` if `hasAuth` is `true`, otherwise it would need to be `null`.

Reply:
Had some difficulty getting the form to correctly load saved connectors with `hasAuth: false` but it should be working now. Connector API will now throw an error if `hasAuth` is `false` and `authType` is anything but `null` or `undefined`.